### PR TITLE
Improve IIS error when wrong application path

### LINKED ISSUE #46021: HTTP Error 500.31 is rather unhelpful
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I've been developing a .NET 6/7 WebAPI site and ran into a HTTP Error 500.31 error when hosting the site in IIS. 

more specifically IIS reports this error page: 

> **HTTP Error 500.31 - Failed to load ASP.NET Core runtime**
> Common solutions to this issue:
> The specified version of Microsoft.NetCore.App or Microsoft.AspNetCore.App was not found.

The ACTUAL problem i had, was that the Web.Config was referencing an incorrect DLL :

> <aspNetCore processPath="dotnet" arguments="DLL-PATH">

the 500.31 error suggest that the framework installation on the host machine is to blame, which sends the developer out on the wrong direction of problem solving. 

### Expected Behavior

I would expect a better error message, that the Path of the DLL (the argument) is incorrect. 

It shouldn't be that hard for dotnet.exe to tell me that the argument is wrong. 

### Steps To Reproduce

- Spin up a new .NET 6 WebAPI site in Visual Studio 2022
- Add Web.Config: 

```
<?xml version="1.0" encoding="utf-8"?>
<configuration>

<location path="." inheritInChildApplications="false">
	<system.webServer>
		<handlers>
			<add name="aspNetCore"
				 path="*"
				 verb="*"
				 modules="AspNetCoreModuleV2"
				 resourceType="Unspecified" />
		</handlers>
		<aspNetCore processPath="dotnet"
					arguments=".\INCORRECT-NAME-OF-DLL.dll"
					stdoutLogEnabled="true"
					stdoutLogFile=".\logs\stdout"
					hostingModel="inprocess" />
	</system.webServer>
</location>
</configuration>
```

- Add new Site in IIS 
- Point to the /Bin/Debug/.NET6.0/ folder
- Start up the site 
- Access the /Swagger page 
- Be met with a HTTP 500.31 error 

### Exceptions (if any)

_No response_

### .NET Version

6 or 7 

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by BrennanConroy**
Fixes https://github.com/dotnet/aspnetcore/issues/46021 by using the hostfxr error code to detect when the application either is missing or not a .dll or .exe file.

Not sure if we should use 500.31 here or add a new error code, or reuse [500.38 Application DLL Not Found](https://learn.microsoft.com/aspnet/core/test/troubleshoot-azure-iis?view=aspnetcore-7.0#50038-ancm-application-dll-not-found).

The error code is defined at https://github.com/dotnet/runtime/blob/fd9b3d101096bf0d8b65fa2cc7ebe6e84d2d280a/src/native/corehost/error_codes.h#L35 and used by hostfxr at https://github.com/dotnet/runtime/blob/fd9b3d101096bf0d8b65fa2cc7ebe6e84d2d280a/src/native/corehost/fxr/command_line.cpp#L161-L181

### CODE REVIEW

File: src/Servers/IIS/AspNetCoreModuleV2/CommonLib/HostFxr.h [hunk 1]
```diff
@@ -30,6 +30,8 @@ typedef int(*hostfxr_get_runtime_property_value_fn)(void* host_context_handle, P
 typedef int(*hostfxr_run_app_fn)(void* host_context_handle);
 typedef int(*hostfxr_close_fn)(void* hostfxr_context_handle);
 
+#define APPARGNOTRUNNABLE 0x80008094
```
Comment (adityamandaleeka) [inline src/Servers/IIS/AspNetCoreModuleV2/CommonLib/HostFxr.h:33]: Use a const instead?

**Comment by adityamandaleeka** [comment]
> or reuse [500.38 Application DLL Not Found](https://learn.microsoft.com/aspnet/core/test/troubleshoot-azure-iis?view=aspnetcore-7.0#50038-ancm-application-dll-not-found).

Based on that doc, it seems like that code was intended specifically for the single file case? Or is the doc trying to say that it _can_ occur in the single file case when run in in-proc mode (but can also occur when not run in in-proc)?

Either way, it also seems like it's specifically about loading a dependency (and not the main executable).

File: src/Servers/IIS/AspNetCoreModuleV2/CommonLib/HostFxr.h (hunk 1 above)
Comment (BrennanConroy) [reply src/Servers/IIS/AspNetCoreModuleV2/CommonLib/HostFxr.h:33]: @elinor-fung is this error code from hostfxr safe for us to rely on?
https://github.com/dotnet/runtime/blob/fd9b3d101096bf0d8b65fa2cc7ebe6e84d2d280a/src/native/corehost/fxr/command_line.cpp#L161-L181

Comment (elinor-fung) [reply src/Servers/IIS/AspNetCoreModuleV2/CommonLib/HostFxr.h:33]: For `hostfxr_get_native_search_directories` (which looks like the case here), yes. We explicitly test that passing in args pointing to an app that doesn't exist for that API should return `AppArgNotRunnable`.

For other APIs (for example `hostfxr_main_startupinfo`), we fall back to trying to do things with the SDK and would end up with a different error code.

**Review by adityamandaleeka — APPROVED** [review]

**Pull request merged**